cmd/boost/deal_status_cmd.go

```diff
@@ -0,0 +1,178 @@
+		},
+	},
+	Before: before,
+	Action: func(cctx *cli.Context) error {
+		ctx := bcli.ReqContext(cctx)
+
+		dealUUID, err := uuid.Parse(cctx.String("deal-uuid"))
+		if err != nil {
+			return err
+		}
+
+		sdir, err := homedir.Expand(cctx.String("repo"))
+		if err != nil {
+			return err
+		}
+
+		n, err := clinode.Setup(ctx, sdir)
+		if err != nil {
+			return err
+		}
+
+		api, closer, err := lcli.GetGatewayAPI(cctx)
+		if err != nil {
+			return fmt.Errorf("cant setup gateway connection: %w", err)
+		}
+		defer closer()
+
+		walletAddr, err := n.Wallet.GetDefault()
+		if err != nil {
+			return err
+		}
+
+		log.Debugw("selected wallet", "wallet", walletAddr)
+
+		maddr, err := address.NewFromString(cctx.String("provider"))
+		if err != nil {
+			return err
+		}
+		minfo, err := api.StateMinerInfo(ctx, maddr, chain_types.EmptyTSK)
+		if err != nil {
+			return err
+		}
+		if minfo.PeerId == nil {
+			return fmt.Errorf("storage provider %s has no peer ID set on-chain", maddr)
+		}
+
+		var maddrs []multiaddr.Multiaddr
+		for _, mma := range minfo.Multiaddrs {
+			ma, err := multiaddr.NewMultiaddrBytes(mma)
+			if err != nil {
+				return fmt.Errorf("storage provider %s had invalid multiaddrs in their info: %w", maddr, err)
+			}
+			maddrs = append(maddrs, ma)
+		}
+		if len(maddrs) == 0 {
+			return fmt.Errorf("storage provider %s has no multiaddrs set on-chain", maddr)
+		}
+
+		log.Debugw("found storage provider", "id", *minfo.PeerId, "multiaddr", maddrs)
+
+		providerStr := cctx.String("provider")
+		minerAddr, err := address.NewFromString(providerStr)
+		if err != nil {
+			return fmt.Errorf("invalid storage provider address '%s': %w", providerStr, err)
+		}
+
+		addrInfo := &peer.AddrInfo{
+			ID:    *minfo.PeerId,
+			Addrs: maddrs,
+		}
+
+		if err := n.Host.Connect(ctx, *addrInfo); err != nil {
+			return fmt.Errorf("failed to connect to peer %s: %w", addrInfo.ID, err)
+		}
+
+		log.Debugw("storage provider on-chain address", "addr", minerAddr)
+
+		retryOpts := lp2pimpl.RetryParameters(time.Second, 1*time.Minute, 5, 5)
+		dc := lp2pimpl.NewDealClient(n.Host, walletAddr, n.Wallet, retryOpts)
+
+		resp, err := dc.SendDealStatusRequest(ctx, *minfo.PeerId, dealUUID)
+		if err != nil {
+			return fmt.Errorf("send deal status request failed: %w", err)
+		}
+
+		msg := "got deal status response"
+		msg += "\n"
+
+		if resp.Error != "" {
+			msg += fmt.Sprintf("  error: %s\n", resp.Error)
+			fmt.Println(msg)
+
+			return nil
+		}
+
+		msg += fmt.Sprintf("  deal uuid: %s\n", resp.DealUUID)
+		msg += fmt.Sprintf("  deal status: %s\n", statusMessage(resp))
+		msg += fmt.Sprintf("  deal label: %s\n", resp.DealStatus.Proposal.Label)
+		msg += fmt.Sprintf("  publish cid: %s\n", resp.DealStatus.PublishCid)
+		msg += fmt.Sprintf("  chain deal id: %d\n", resp.DealStatus.ChainDealID)
+		fmt.Println(msg)
+
+		return nil
+	},
+}
+
+// statusMessage is based on dealResolver.Message
+func statusMessage(resp *types.DealStatusResponse) string {
```

Comment:
Let's move this to common code between the resolver and here

Reply:
The data model is completely different, even if the logic is the same, so this might be hard, but will look into it.

Reply:
Ok not a big deal, if it doesn't make sense let's leave it as-is